tensorflow/java/src/main/java/org/tensorflow/op/training/Gradients.java

```diff
@@ -0,0 +1,153 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
```

Comment:
Why `training/`? Seems like `gradients` could be a top level operation too? One can use gradients for things other than training :)

Reply:
No problem. Right now, the `Op` classification is still obscure to me, I'm taking some guesses here and there but we will probably need to settle this at some point?

Reply:
BTW, the Python code was recently updated to generate symbols in finer grained namespaces (https://github.com/tensorflow/tensorflow/commit/c1ff1164e30186d847f7d4f9e9ce5d40936a2c1c#diff-d75a97d0f69d6f87ab79be6e2423d87b). So perhaps we could use the same for Java (without the baggage of backward compatibility. For example, `Acos` can be just in the `math` namespace and not in the top level one).

If you're enthusiastic, I'd be more than happy to see a PR adding Java API defs :)